central/resourcecollection/datastore/datastore_impl_test.go

```diff
@@ -323,8 +325,278 @@ func (s *CollectionPostgresDataStoreTestSuite) TestCollectionWorkflows() {
+						{
+							FieldName: pkgSearch.ClusterRole.String(),
+							Operator:  storage.BooleanOperator_OR,
+						},
+					},
+				},
+			},
+			nil,
+		},
+		{
+			"unknown selector rule field name",
+			[]*storage.ResourceSelector{
+				{
+					Rules: []*storage.SelectorRule{
+						{
+							FieldName: "bad label",
+							Operator:  storage.BooleanOperator_OR,
+						},
+					},
+				},
+			},
+			nil,
+		},
+		{
+			"nil resource selector list",
+			nil,
+			[]*v1.Query{},
+		},
+		{
+			"empty resource selector list",
+			[]*storage.ResourceSelector{},
+			[]*v1.Query{},
+		},
+		{
+			"two entry resource selector list",
+			[]*storage.ResourceSelector{
+				{
+					Rules: []*storage.SelectorRule{
+						{
+							FieldName: pkgSearch.Cluster.String(),
+							Operator:  storage.BooleanOperator_OR,
+							Values: []*storage.RuleValue{
+								{
+									Value: "1",
+								},
+								{
+									Value: "2",
+								},
+							},
+						},
+					},
+				},
+				{
+					Rules: []*storage.SelectorRule{
+						{
+							FieldName: pkgSearch.Namespace.String(),
+							Operator:  storage.BooleanOperator_OR,
+							Values: []*storage.RuleValue{
+								{
+									Value: "3",
+								},
+							},
+						},
+					},
+				},
+			},
+			[]*v1.Query{
+				{
+					Query: &v1.Query_Disjunction{
+						Disjunction: &v1.DisjunctionQuery{
+							Queries: []*v1.Query{
+								{
+									Query: getBaseQuery(pkgSearch.Cluster, "r/1"),
+								},
+								{
+									Query: getBaseQuery(pkgSearch.Cluster, "r/2"),
+								},
+							},
+						},
+					},
+				},
+				{
+					Query: getBaseQuery(pkgSearch.Namespace, "r/3"),
+				},
+			},
+		},
+		{
+			"nil selector rule list",
+			[]*storage.ResourceSelector{
+				{
+					Rules: nil,
+				},
+			},
+			[]*v1.Query{},
+		},
+		{
+			"empty selector rule list",
+			[]*storage.ResourceSelector{
+				{
+					Rules: []*storage.SelectorRule{},
+				},
+			},
+			[]*v1.Query{},
+		},
+		{
+			"and selector rule operator",
+			[]*storage.ResourceSelector{
+				{
+					Rules: []*storage.SelectorRule{
+						{
+							FieldName: pkgSearch.Cluster.String(),
+							Operator:  storage.BooleanOperator_AND,
+						},
+					},
+				},
+			},
+			nil,
+		},
+	}
+
+	for _, test := range collectionToQueryTests {
+		s.T().Run(test.name, func(t *testing.T) {
+			testCollection := getTestCollection("1", nil)
+			testCollection.ResourceSelectors = test.resourceSelectors
+			parsedQueries, err := collectionToQueries(testCollection)
+			if test.expectedQueries == nil {
+				assert.Error(t, err)
+				assert.Nil(s.T(), parsedQueries)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, len(test.expectedQueries), len(parsedQueries))
+				for i := 0; i < len(test.expectedQueries) && i < len(parsedQueries); i++ {
+					assert.Equal(t, test.expectedQueries[i], parsedQueries[i])
+				}
+			}
+		})
+	}
+}
+
+// func (s *CollectionPostgresDataStoreTestSuite) TestResolveCollectionQuery() {
```

Comment:
I can't uncomment it without failing style checks, it requires the function to be included in the datastore definition, and style checks will fail because "it appears the datastore is storing *v1.Query objects"